irida_staramr_results/irida_api.py

```diff
@@ -112,112 +110,169 @@ def _create_session(self):
+
+        response = self._get_resource(target_url)
+
+        if target_dict:  # we are targeting specific resources in the response
+
+            try:
+                resources_list = response.json()["resource"]["resources"]
+            except KeyError as e:
+                # TODO: This try except block has been added to log a crash that has occurred, to find the source.
+                #   This is occurring for an unknown reason.
+                #   Once docs can be gathered displaying information, we can determine the source of the bug and fix it.
+                logging.error("Dumping json response from IRIDA:")
+                logging.error(str(response.json()))
+                logging.error("Dumping python KeyError:")
+                logging.error(e)
+                error_txt = "Response from IRIDA Could not be parsed. Please show the log to your IRIDA Administrator."
+                logging.error(error_txt)
+                raise exceptions.IridaKeyError(error_txt)
+
+            # try to get all keys from target_dict to our list or links
+            try:
+                links_list = next(
+                    r["links"] for r in resources_list
+                    if r[target_dict["key"]].lower() == str(target_dict["value"]).lower()
+                )
+
+            except KeyError:
+                raise exceptions.IridaKeyError(target_dict["key"] + " not found. Available keys: "
+                                                                    ", ".join(resources_list[0].keys()))
 
-    def get_project_from_id(self, _id):
-        """ Returns a project resource object from an id of a project """
-        return self._get_resource_from_path(f"/projects/{_id}")
+            except StopIteration:
+                raise exceptions.IridaKeyError(target_dict["value"] + " not found.")
 
-    def get_analyses_from_projects(self, _id):
+
+        else:
+            links_list = response.json()["resource"]["links"]
+
+        # find the right "href" based on "rel"/target_key
+        try:
+            ret_val = next(link["href"] for link in links_list
+                           if link["rel"] == target_key)
+        except StopIteration:
+            error_txt = target_key + " not found in links. Available links: " +\
+                        ", ".join([str(link["rel"]) for link in links_list])
+            logging.debug(error_txt)
+            raise exceptions.IridaKeyError(error_txt)
+
+        return ret_val
+
+    def get_amr_analysis_results(self, project_id):
+        """
+        Get all AMR detection analysis results from a project id.
+        If nothing is found, it returns an empty array.
+        :param project_id:
+        :return analysis_result_list:
         """
-        Returns an array containing analysis objects for a given project
 
-        :param _id:
+        try:
+            project_analysis_submissions = self._get_analysis_submissions_from_projects(project_id)
+        except KeyError:
```

Comment:
Since we indicated in the function comment that an empty array will be returned if nothing is found, we should return `[]` instead of raising an `IridaKeyError` here.

Reply:
this IridaKeyError is for catching project Id not existing. I can change the function comment to 

```
Get all AMR detection analysis results from a project id.
If no analysis results found in the project, it returns an empty array.
```